File: tests/fixtures/sample_project_ruby/src/db/connection.rb
# Behaviour every backing store must provide.
class DataStore
  # Run a statement and return the number of affected rows.
  def execute(sql)
    0
  end
end

# A pooled connection to the fixture database.
class Connection < DataStore
  # Return a process-wide connection to the fixture database.
  def self.get_connection
    Connection.new
  end

  # Close the connection and release the pool slot.
  def close
    true
  end

  def execute(sql)
    sql.length
  end

  # Create the fixture tables if they don't exist yet.
  def self.ensure_schema(conn)
    conn.execute("CREATE TABLE users") > 0
  end
end

File: tests/fixtures/sample_project_php/src/Db/Connection.php
<?php

namespace Acme\Db;

/** Behaviour every backing store must provide. */
interface DataStore
{
    /** Run a statement and return the number of affected rows. */
    public function execute($sql);
}

/** A pooled connection to the fixture database. */
class Connection implements DataStore
{
    /** The backing database URL. */
    public $url;

    /** Return a process-wide connection to the fixture database. */
    public static function getConnection()
    {
        return new Connection();
    }

    /** Close the connection and release the pool slot. */
    public function close()
    {
        return true;
    }

    public function execute($sql)
    {
        return strlen($sql);
    }

    /** Create the fixture tables if they don't exist yet. */
    public static function ensureSchema($conn)
    {
        return $conn->execute("CREATE TABLE users") > 0;
    }
}
